src/openms/include/OpenMS/FILTERING/TRANSFORMERS/LinearResamplerAlign.h

```diff
@@ -186,13 +207,93 @@ namespace OpenMS
+        to the intensities in the output container (which in most cases will be
+        zero).
+
+        The intensities will be distributed between the two closest resampling
+        points, thus conserving the sum of intensity over the whole container.
+
+        Note that all intensity in the input data container that is in peaks
+        outside the range of the output container will simply be added to the
+        first or last data point.
+
+        @param mz_raw_it Start of the input container to be resampled (containing the m/z data)
+        @param mz_raw_end End of the input container to be resampled (containing the m/z data)
+        @param int_raw_it Start of the input container to be resampled (containing the intensity data)
+        @param int_raw_end End of the input container to be resampled (containing the intensity data)
+        @param mz_resample_it Iterator pointing to start of the output spectrum range (m/z which need to be populated)
+        @param mz_resample_it Iterator pointing to end of the output spectrum range (m/z which need to be populated)
+        @param int_resample_it Iterator pointing to start of the output spectrum range (intensities)
+        @param int_resample_it Iterator pointing to end of the output spectrum range (intensities)
+
+    */
+    template <typename PeakTypeIterator, typename ConstPeakTypeIterator>
+    void raster(ConstPeakTypeIterator mz_raw_it, ConstPeakTypeIterator mz_raw_end,
+        ConstPeakTypeIterator int_raw_it, ConstPeakTypeIterator int_raw_end,
+        PeakTypeIterator mz_resample_it, PeakTypeIterator mz_resample_end,
+        PeakTypeIterator int_resample_it, PeakTypeIterator int_resample_end
+        )
+    {
+      OPENMS_PRECONDITION(mz_resample_it != mz_resample_end, "Output iterators cannot be identical") // as we use +1
+      OPENMS_PRECONDITION(std::distance(mz_resample_it, mz_resample_end) == std::distance(int_resample_it, int_resample_end),
+          "Resample m/z and intensity iterators need to cover the same distance")
+      OPENMS_PRECONDITION(std::distance(mz_raw_it, mz_raw_end) == std::distance(int_raw_it, int_raw_end),
+          "Raw m/z and intensity iterators need to cover the same distance")
+      // OPENMS_PRECONDITION(raw_it != raw_end, "Input iterators cannot be identical")
+
+      PeakTypeIterator mz_resample_start = mz_resample_it;
+
+      // need to get the raw iterator between two resampled iterators of the raw data
+      while (mz_raw_it != mz_raw_end && (*mz_raw_it) < (*mz_resample_it) )
+      {
+        (*int_resample_it) = *int_resample_it + *int_raw_it;
+        ++mz_raw_it;
+        ++int_raw_it;
+      }
+
+      while (mz_raw_it != mz_raw_end)
+      {
+        //advance the resample iterator until our raw point is between two resampled iterators
+        while (mz_resample_it != mz_resample_end && *mz_resample_it < *mz_raw_it) {++mz_resample_it; ++int_resample_it;}
```

Comment:
maybe break the block into several lines (I thought we usually only encourage single line blocks if there is only one expression in it).